README.md

```diff
@@ -1,4 +1,285 @@
+  }).isRequired
+};
+```
+If `SubComponent` will require `addEntity` in the future, all we need to do is
+declaring the new dependency in its `SubComponent.propTypes`, without having to
+change the actual component code (and tests!). Note that is totally reasonable
+and encouraged to declare shapes with only one field.
+
+A shortcut is also available when reshaping is required for propagating
+properties to subcomponents. The parent component can declare that it requires
+a [`shapeup.reshapeFunc`](#reshapeFunc) property as part of the shape, which
+can then be used to reshape the object implementing the shape itself. So, the
+example above can be rewritten as:
+
+```javascript
+// In parent-component.js.
+class ParentComponent extends React.Component {
+  ...
+  <SubComponent
+    // The api object has a reshape method. Use that to create an object with
+    // the shape required by SubComponent starting from this.props.api.
+    api={this.props.api.reshape(SubComponent.propTypes.api)}
+  />
+  ...
+}
+ParentComponent.propTypes = {
+  api: shapeup.shape({
+    addEntity: PropTypes.func.isRequired,
+    removeEntity: PropTypes.func.isRequired,
+    reshape: shapeup.reshapeFunc
+  }).isRequired
+};
+
+// In subcomponent.js.
+SubComponent.propTypes = {
+  api: shapeup.shape({
+    removeEntity: PropTypes.func.isRequired
+  }).isRequired
+};
+```
+The name of the reshape field is not important, the value is, as it declares
+that field to be the placeholder for the reshape function. But how is this
+function provided? It can be provided in two ways:
+- by using [`shapeup.fromShape`](#fromShape) as described above: when building
+  the object, [`shapeup.fromShape`](#fromShape) includes a proper
+  implementation of the reshape function if the given shape requires it;
+- by wrapping the object provided as shape with
+  [`shapeup.addReshape`](#addReshape), in case
+  [`shapeup.fromShape`](#fromShape) is not used, and the object implementing
+  the shape is manually built. For instance:
+```javascript
+<MyComponent
+  api={shapeup.addReshape({
+    getById: someobj.getById.bind(someobj),
+    getAll: someobj.getAll.bind(someobj)
+  })}
+/>
+```
+
+Reshaping is the preferred way of propagating properties in deeply nested
+component trees when using shapeup, as it allows extending the properties by
+only updating the initial object and the `propTypes` declaration of components,
+rather than updating how every single component in the tree is instantiated.
+
+## Frozen shapes
+
+As with the traditional prop-types, it is possible to chain
+[`shapeup.shape`](#shape) with `isRequired`, in order to make that property
+required. It is also possible to chain the declaration with `frozen` in order
+to ensure that the provided property is deeply frozen. This is useful when
+providing a shape and willing to avoid the usual problems of object mutation,
+like unwanted side effects, bugs that are difficult to track down, unrequired
```

Comment:
providing a shape and wanting to avoid the usual problems of object mutation, such as unwanted side effects...

Reply:
Done.